clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -37,44 +38,43 @@
  *
  * @param <T> Return type of the result (Can be Void if there is no response)
  */
-public class RequestFuture<T> {
-
-    private boolean isDone = false;
-    private T value;
-    private RuntimeException exception;
-    private List<RequestFutureListener<T>> listeners = new ArrayList<>();
+public class RequestFuture<T> implements ConsumerNetworkClient.PollCondition {
 
+    private final AtomicBoolean isDone = new AtomicBoolean(false);
+    private final AtomicReference<T> value = new AtomicReference<>();
+    private final AtomicReference<RuntimeException> exception = new AtomicReference<>();
```

Comment:
I think the usage is simple enough here that we can go with volatiles for better memory and less verbosity.